sechub-wrapper-owasp-zap/src/main/java/com/mercedesbenz/sechub/zapwrapper/internal/scan/ClientApiFacade.java

```diff
@@ -370,18 +372,34 @@ public ApiResponse stopSpiderScan(String scanId) throws ClientApiException {
     }
 
     /**
-     * Get a list of all URLs detected by the spider scan.
+     * For the given scanId read all the spider results and add them to a list. Each
+     * entry of the list consists of a Map with the form: {processed=true,
+     * statusReason=OK, method=GET, reasonNotProcessed=, messageId=6,
+     * url=http://example.com, statusCode=200}
      *
-     * @return api response of ZAP
-     * @throws ClientApiException when anything goes wrong communicating with ZAP
+     * @param scanId
+     * @param zapProductMessageHelper
+     * @throws ClientApiException
      */
-    public List<String> getAllSpiderUrls() throws ClientApiException {
-        List<ApiResponse> results = ((ApiResponseList) clientApi.spider.allUrls()).getItems();
-        List<String> urls = new ArrayList<>();
-        for (ApiResponse response : results) {
-            urls.add(response.toString());
+    public List<Map<String, ApiResponse>> getFullSpiderResults(String scanId) throws ClientApiException {
+        List<Map<String, ApiResponse>> fullResults = new LinkedList<>();
+
+        ApiResponseList results = (ApiResponseList) clientApi.spider.fullResults(scanId);
```

Comment:
ZAP has two different spiders:
1. "normal"
2. ajax spider

do we get here all results?